cmake/EthCompilerSettings.cmake

```diff
@@ -266,3 +266,13 @@ option(USE_Z3 "Allow compiling with Z3 SMT solver integration" ON)
 if(UNIX AND NOT APPLE)
 	option(USE_Z3_DLOPEN "Dynamically load the Z3 SMT solver instead of linking against it." OFF)
 endif()
+
+# Custom linker
+# TODO: This mechanism should be replaced by setting CMAKE_LINKER_TYPE once the CMake minimum version reaches 3.29
+if(DEFINED USE_LD AND USE_LD)
+	if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang"))
+		add_link_options("-fuse-ld=${USE_LD}")
+	else ()
+		message(FATAL_ERROR "Custom linker only supported for gcc and clang.")
+	endif ()
+endif()
```

Comment:
Right, `CMAKE_EXE_LINKER_FLAGS` should work if there is shared libraries or module libraries - which I guess we don't have. Otherwise there'd also be `CMAKE_SHARED_LINKER_FLAGS` and `CMAKE_MODULE_LINKER_FLAGS` which have to be set, in that case I think the `add_link_options` would be nicer. Let me change it to `CMAKE_EXE_LINKER_FLAGS`, though!

Reply:
I mean, technically ``add_link_options`` is also rather evil depending on its scope - what we'd really need would be a nice ``solc_settings`` or ``solc-base`` target for which one could set ``PUBLIC`` ``target_link_options`` and then have everything else link to that, but that'd mean rewriting half of our rather outdated cmake logic, which we never bothered to do so far, since it never seemed like the best use of time...

Reply:
Right.. or bump CMake minimum version to `3.29` and use `CMAKE_LINKER_TYPE` - although that is probably not an option right now and arguably an explicit per-target selection would still be nicer. :)